src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -68,27 +76,57 @@ class CDCL(private var clauses: MutableList<MutableList<Int>>, private val varsN
     // list of unit clauses to propagate
     private val units: MutableList<Int> = mutableListOf()
 
+    private var assumptions: List<Int> = emptyList()
+
+    private fun clearTrail(until: Int = -1) {
+        while (trail.isNotEmpty() && vars[trail.last()].level > until) {
+            delVariable(trail.removeLast())
+        }
+    }
+
+    fun solveWithAssumptions(currentAssumptions: List<Int> = emptyList()): List<Int>? {
+        assumptions = currentAssumptions
+        val result = solve()
+        assumptions.forEach {
+            if (getStatus(it) == VarStatus.FALSE) {
+                return null
+            }
+        }
+        return  result
+    }
```

Comment:
Shouldn't we clear assumptions before return? In case we call solve right after solveWithAssumptions

Reply:
Really